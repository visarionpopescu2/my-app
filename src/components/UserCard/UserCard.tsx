import { USER_DETAILS } from 'types';
import {Card, Col, Image, Row, Typography} from 'antd';

function UserCard(props: { user: USER_DETAILS | undefined }) {
  const { user } = props;
  
  const buildUserDetails = () => {
    if (!user) {
      return (
        <Col>
          <Typography.Text type="danger">No user has been found</Typography.Text>
        </Col>
      );
    }
    
    return (
      <>
        <Col xs={24} md={12} lg={10}>
          <Typography.Title level={5} style={{marginTop: 0}}>
            Contact Details
          </Typography.Title>
          <Typography.Paragraph>
            <strong>Name:</strong> {user.name}
          </Typography.Paragraph>
          <Typography.Paragraph>
            <strong>Username:</strong> {user.username}
          </Typography.Paragraph>
          <Typography.Paragraph>
            <strong>Email address:</strong> {user.email}
          </Typography.Paragraph>
          <Typography.Paragraph>
            <strong>Phone number:</strong> {user.phone}
          </Typography.Paragraph>
        </Col>
  
        <Col xs={24} md={12} lg={10}>
          <Typography.Title level={5} style={{marginTop: 0}}>
            Address
          </Typography.Title>
          <Typography.Paragraph>
            <strong>City:</strong> {user.address.city}
          </Typography.Paragraph>
          <Typography.Paragraph>
            <strong>Street:</strong> {user.address.street}
          </Typography.Paragraph>
          <Typography.Paragraph>
            <strong>Suite:</strong> {user.address.suite}
          </Typography.Paragraph>
          <Typography.Paragraph>
            <strong>Zip code:</strong> {user.address.zipcode}
          </Typography.Paragraph>
        </Col>
      </>
    );
  }

  return (
    <Card>
      <Row gutter={16}>
        <Col xs={24} lg={4}>
          <div style={{ maxWidth: '110px' }}>
            <Image src={process.env.REACT_APP_USER_IMG_URL} alt="User picture" />
          </div>
        </Col>
  
        {buildUserDetails()}
      </Row>
    </Card>
  )
}

export default UserCard;
