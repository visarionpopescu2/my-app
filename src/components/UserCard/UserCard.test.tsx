import { render } from '@testing-library/react';
import UserCard from './UserCard';

const USER = {
  id: '1',
  name: 'name',
  username: 'username',
  email: 'email',
  phone: '123445',
  website: 'www.website.com',
  company: { name: 'company', bs: 'business', catchPhrase: 'description' },
  address: { street: 'street', suite: 'suite', zipcode: '1234', city: 'city', geo: { lat: '11', lng: '22' }}
};

describe('UserCard', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  
  it('should render UserCard component when has user as prop', () => {
    let view = render(<UserCard user={USER} />);
    expect(view.asFragment()).toMatchSnapshot();
  });
  
  it('should render UserCard component when does not have user as prop', () => {
    let view = render(<UserCard user={undefined} />);
    expect(view.asFragment()).toMatchSnapshot();
  })
});
