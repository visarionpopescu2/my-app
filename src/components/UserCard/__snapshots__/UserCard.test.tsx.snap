// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should render UserCard component when does not have user as prop 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-2i2tap"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row css-dev-only-do-not-override-2i2tap"
        style="margin-left: -8px; margin-right: -8px;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-lg-4 css-dev-only-do-not-override-2i2tap"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <div
            style="max-width: 110px;"
          >
            <div
              class="ant-image css-dev-only-do-not-override-2i2tap"
            >
              <img
                alt="User picture"
                class="ant-image-img css-dev-only-do-not-override-2i2tap"
                src="https://shorturl.at/dzBW2"
              />
              <div
                class="ant-image-mask"
              >
                <div
                  class="ant-image-mask-info"
                >
                  <span
                    aria-label="eye"
                    class="anticon anticon-eye"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      />
                    </svg>
                  </span>
                  Preview
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-col css-dev-only-do-not-override-2i2tap"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <span
            class="ant-typography ant-typography-danger css-dev-only-do-not-override-2i2tap"
          >
            No user has been found
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`UserCard should render UserCard component when has user as prop 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-2i2tap"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-row css-dev-only-do-not-override-2i2tap"
        style="margin-left: -8px; margin-right: -8px;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-lg-4 css-dev-only-do-not-override-2i2tap"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <div
            style="max-width: 110px;"
          >
            <div
              class="ant-image css-dev-only-do-not-override-2i2tap"
            >
              <img
                alt="User picture"
                class="ant-image-img css-dev-only-do-not-override-2i2tap"
                src="https://shorturl.at/dzBW2"
              />
              <div
                class="ant-image-mask"
              >
                <div
                  class="ant-image-mask-info"
                >
                  <span
                    aria-label="eye"
                    class="anticon anticon-eye"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      />
                    </svg>
                  </span>
                  Preview
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-md-12 ant-col-lg-10 css-dev-only-do-not-override-2i2tap"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <h5
            class="ant-typography css-dev-only-do-not-override-2i2tap"
            style="margin-top: 0px;"
          >
            Contact Details
          </h5>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Name:
            </strong>
             name
          </div>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Username:
            </strong>
             username
          </div>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Email address:
            </strong>
             email
          </div>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Phone number:
            </strong>
             123445
          </div>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-md-12 ant-col-lg-10 css-dev-only-do-not-override-2i2tap"
          style="padding-left: 8px; padding-right: 8px;"
        >
          <h5
            class="ant-typography css-dev-only-do-not-override-2i2tap"
            style="margin-top: 0px;"
          >
            Address
          </h5>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              City:
            </strong>
             city
          </div>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Street:
            </strong>
             street
          </div>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Suite:
            </strong>
             suite
          </div>
          <div
            class="ant-typography css-dev-only-do-not-override-2i2tap"
          >
            <strong>
              Zip code:
            </strong>
             1234
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
